pkg/machine/compression/xz.go

```diff
@@ -0,0 +1,87 @@
+package compression
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/sirupsen/logrus"
+	"github.com/ulikunitz/xz"
+)
+
+type xzDecompressor struct {
+	compressedFilePath string
+	compressedFile     *os.File
+}
+
+func newXzDecompressor(compressedFilePath string) decompressor {
+	return &xzDecompressor{
+		compressedFilePath: compressedFilePath,
+	}
+}
+
+func (d *xzDecompressor) srcFilePath() string {
+	return d.compressedFilePath
+}
+
+func (d *xzDecompressor) reader() (io.Reader, error) {
+	srcFile, err := os.Open(d.compressedFilePath)
+	if err != nil {
+		return nil, err
+	}
+	d.compressedFile = srcFile
+	return srcFile, nil
+}
+
+// Will error out if file without .Xz already exists
+// Maybe extracting then renaming is a good idea here..
+// depends on Xz: not pre-installed on mac, so it becomes a brew dependency
+func (*xzDecompressor) copy(w *os.File, r io.Reader) error {
+	var cmd *exec.Cmd
+	var read io.Reader
+
+	// Prefer Xz utils for fastest performance, fallback to go xi2 impl
+	if _, err := exec.LookPath("xz"); err == nil {
```

Comment:
We should move to zstd.

Reply:
This PR is about cleaning up code inconsistencies and duplication in [decompress.go](https://github.com/containers/podman/blob/main/pkg/machine/compression/decompress.go). I am trying not to add or remove anything. That's on purpose. If supports for `xz` should be removed it will be in a separate PR.